manifest/manifest.go

```diff
@@ -0,0 +1,150 @@
+package manifest
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/filecoin-project/go-f3/certs"
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/ipfs/go-datastore"
+	"golang.org/x/xerrors"
+)
+
+type OnManifestChange func(ctx context.Context, initialInstance uint64, rebootstrap bool, errCh chan error)
+
+type ManifestProvider interface {
+	// Run starts any background tasks required for the operation
+	// of the manifest provider.
+	Run(context.Context, chan error)
+	// Returns the list of gpbft options that should be used
+	GpbftOptions() []gpbft.Option
+	// suscribe to manifest updates
+	Subscribe() <-chan struct{}
+	// Set callback for manifest changes
+	SetManifestChangeCb(OnManifestChange)
```

Comment:
Without the comment above this function it would have taken me loner than it should to infer `Cb` is short for "Callback". 
* I recommend a clearer name and documentation of semantics (similar to what I recommended for `Subscribe`)
* At API design level, it seems odd to have a subscription mechanism to receive notifications, _and_ a hook registry to execute once manifest changes. I am curious if both are necessary and whether the API surface here can be reduced.

Reply:
I think you may be right, it is not clear yet if I'll need the `Subscribe()`  in the end. Before I needed it to notify the runner that there was a new manifest so it can pick up the new configuration. Now that you mention it, I may be able to integrate this as part of the F3 callback. Unfortunately, that logic comes in the next PR, so if you don't mind, let's revisit this one with the whole logic implemented in the next PR. Thanks!